go/vt/vtctl/grpcvtctldserver/server.go

```diff
@@ -364,16 +364,41 @@ func (s *VtctldServer) ApplyVSchema(ctx context.Context, req *vtctldatapb.ApplyV
 		vs = req.VSchema
 	}
 
+	ksVs, err := vindexes.BuildKeyspace(vs, s.ws.SQLParser())
```

Comment:
This should come after the dry-run check, no? I wouldn't expect work to be done on a dry-run and we don't handle the error in that case either.

Reply:
Ah, well that is how things used to work before this PR. I thought that it would be useful to report unknown params even during dry run, but in order to do that I needed to move things around a bit. This is equivalent to work we previously did here https://github.com/vitessio/vitess/pull/13322/files#diff-ad37010accc5a29c6751d18124328bb882457e8b2527db688b854c81fe23861cR3339-R3360